crates/daemon/src/monitor.rs

```diff
@@ -466,66 +485,46 @@ impl Actor {
-        );
-
-        while let Some(event) = ready_events.pop() {
-            match event {
-                Event::LockFinality(id) => {
-                    self.invoke_cfd_command(id, |cfd| Ok(Some(cfd.handle_lock_confirmed())))
-                        .await
-                }
-                Event::CommitFinality(id) => {
-                    self.invoke_cfd_command(id, |cfd| Ok(Some(cfd.handle_commit_confirmed())))
-                        .await
-                }
-                Event::CloseFinality(id) => {
-                    self.invoke_cfd_command(id, |cfd| {
-                        Ok(Some(cfd.handle_collaborative_settlement_confirmed()))
-                    })
-                    .await
-                }
-                Event::CetTimelockExpired(id) => {
-                    self.invoke_cfd_command(id, |cfd| cfd.handle_cet_timelock_expired().map(Some))
-                        .await
-                }
-                Event::CetFinality(id) => {
-                    self.invoke_cfd_command(id, |cfd| Ok(Some(cfd.handle_cet_confirmed())))
-                        .await
-                }
-                Event::RefundFinality(id) => {
-                    self.invoke_cfd_command(id, |cfd| Ok(Some(cfd.handle_refund_confirmed())))
-                        .await
-                }
-                Event::RevokedTransactionFound(id) => {
-                    self.invoke_cfd_command(id, |cfd| Ok(Some(cfd.handle_revoke_confirmed())))
-                        .await
-                }
-                Event::RefundTimelockExpired(id) => {
-                    self.invoke_cfd_command(id, |cfd| cfd.handle_refund_timelock_expired())
-                        .await
+        let (tx_script_updates, rx_script_updates) = tokio::sync::mpsc::channel(BATCH_SIZE * 4);
+
+        let scripts = self
+            .state
+            .monitoring_scripts()
+            .cloned()
+            .collect::<Vec<Script>>();
+
+        let batches = scripts.chunks(BATCH_SIZE).map(|batch| batch.to_owned());
+
+        let client = self.client.clone();
+
+        for batch in batches {
+            tokio::task::spawn_blocking({
+                let client = client.clone();
+                let tx_script_updates = tx_script_updates.clone();
+                move || {
+                    for script in batch {
+                        match client.script_get_history(&script) {
+                            Ok(script_history_response) => {
+                                // We use blocking_send to stay within a sync context here
+                                // One should not use async code in a spawn_blocking block
+                                if let Err(e) =
+                                    tx_script_updates.blocking_send(script_history_response)
+                                {
+                                    tracing::error!(
+                                        "Error when processing script_get_history response: {e:#}"
```

Comment:
🛠 this one gets fired off all the time when I tested it with response "channel closed", and we never stop.

I don't think just logging the errors here is an option....

Reply:
I was unable to directly reproduce the "channel closed" issue, but I found a problem in the implementation where the sender and the receiver were not properly tied together, because the sender was not moved correctly. 
I think that this issue caused this behavior. In the PRs replacing this one I also added tests to ensure that the underlying batch functionality works, but we yet have to test the complete production code again. 

I also thought about adding a test that spins up the complete monitor Actor, but that is a bit too much for now...